js/gantt-helper.js

```diff
@@ -0,0 +1,610 @@
+                  }
+               },
+               { unit: "day", step: 1, format: "%j %D" }
+            ]
+         },
+         {
+            name: "month",
+            scale_height: 50,
+            min_column_width: 120,
+            scales: [
+               { unit: "month", format: "%F, %Y" },
+               { unit: "week", format: "Week #%W" }
+            ]
+         },
+         {
+            name: "quarter",
+            height: 50,
+            min_column_width: 90,
+            scales: [
+               { unit: "month", step: 1, format: "%M" },
+               {
+                  unit: "quarter",
+                  step: 1,
+                  format: function(date) {
+                     var dateToStr = gantt.date.date_to_str("%M");
+                     var endDate = gantt.date.add(gantt.date.add(date, 3, "month"), -1, "day");
+                     return dateToStr(date) + " - " + dateToStr(endDate);
+                  }
+               }
+            ]
+         },
+         {
+            name: "year",
+            scale_height: 50,
+            min_column_width: 30,
+            scales: [
+               { unit: "year", step: 1, format: "%Y" }
+            ]
+         }
+      ]
+   };
+
+   gantt.ext.zoom.init(zoomConfig);
+   gantt.ext.zoom.setLevel("month");
+
+   // <<<<< Configs
+
+
+   // >>>>> Event handlers
+
+   gantt.ext.zoom.attachEvent("onAfterZoom", function(level, config) {
+      document.querySelector(".gantt_radio[value='" + config.name + "']").checked = true;
+   });
+
+   var radios = document.getElementsByName("scale");
+   for (var i = 0; i < radios.length; i++) {
+      radios[i].onclick = function(event) {
+         gantt.ext.zoom.setLevel(event.target.value);
+      };
+   }
+
+   // enable task reordering (on same level) between projects
+   gantt.attachEvent("onBeforeTaskMove", function(id, parent, tindex) {
+      var task = gantt.getTask(id);
+      if (task.parent != parent)
+         return false;
+      return true;
+   });
+
+   if (!readonly) {
+      // catch task drag event to update db
+      gantt.attachEvent("onAfterTaskDrag", function(id, mode, e) {
+         var task = gantt.getTask(id);
+         var progress = (Math.round(task.progress * 100 / 5) * 5) / 100; // prevent server side exception for wrong stepping
+         $.ajax({
+            url,
+            type: 'POST',
+            data: {
+               updateTask: 1,
+               task: {
+                  id: task.linktask_id,
+                  start_date: formatFunc(task.start_date),
+                  end_date: formatFunc(task.end_date),
+                  progress
+               }
+            },
+            success: function(resp) {
+               var json = JSON.parse(resp);
+               if (json.ok) {
+                  task.progress = progress;
+                  gantt.updateTask(task.id);
+               } else {
+                  console.log('Could not update Task[' + id + ']: ' + json.error);
```

Comment:
Error should be show to the user, not in the console, I guess with:
```suggestion
                  gantt.alert('Could not update Task[' + id + ']: ' + json.error);
```

Same remark for all other `console` usages. This will also fix some of the remaining lint issue.